File: php/classes/abstract_classes/abstract_class.php
#!/usr/bin/php8.1
<?php

abstract class Datum {
    public string $description;
}

class Content extends Datum {
    private string $core;

    public function __construct(string $core, string $description = "") {
        $this->core = $core;
        $this->description = $description;
    }

    public function show() : void {
        print("Description: " . $this->description
            . "\nCore: " . $this->core . "\n");
    }
};

// $data = new Datum(); // One cannot instantiate.

$lectio = new Content(
  "In omnibus requiem quaesivi, et nusquam inveni nisi in angulo cum libro.",
  "De beneficiis lectionis"
);

$lectio->show();

File: c++/classes/abstract_classes/abstract_class.cpp
#include <iostream>

using std::string, std::cout, std::endl;

class Datum {
protected:
  string description;

public:
  virtual void show() = 0;

  string formatDescriptionAsText() {
    return ("Description: " + description);
  }
};

class Content: public Datum {
protected:
  string core;

public:
  Content(string content_core, string content_description = "") :
    core(content_core) {
      description = content_description;
  }

  string formatCoreAsText() {
    return ("Core: " + core);
  }

  void show() {
    cout << formatDescriptionAsText() << endl
      << formatCoreAsText() << endl;
  }
};

int main() {
  // Datum data; // One cannot instantiate.

  Content lectio(
    "In omnibus requiem quaesivi, et nusquam inveni nisi in angulo cum libro.",
    "De beneficiis lectionis"
  );

  lectio.show();
}
